components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/revoke/OAuthRevocationEndpoint.java

```diff
@@ -89,12 +95,31 @@ public Response revokeAccessToken(@Context HttpServletRequest request, Multivalu
 
         try {
             startSuperTenantFlow();
+            Map<String, Object> params = new HashMap<>();
+            if (MapUtils.isNotEmpty(paramMap)) {
+                paramMap.forEach((key, value) -> {
+                    if (TOKEN_PARAM.equals(key) && CollectionUtils.isNotEmpty(value)) {
+                        params.put("token", value.get(0).replaceAll(".", "*"));
+                    } else {
+                        params.put(key, value);
+                    }
+                });
+            }
+            if (OAuth2Util.isDiagnosticLogsEnabled()) {
```

Comment:
this should be checked prior to creating params map

Reply:
the params map is being used at L120 as well. Therefore, can't include the param map in the if block. We can always initiate an empty hashMap and add the values after a check. Is that a better approach?